sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -185,8 +183,21 @@ case class DataSource(
     }
 
     SchemaUtils.checkColumnNameDuplication(
```

Comment:
Also, I think we couldn't catch the duplication in case of inferred data schema except for here. So, I put the check here and in the constructor.